paper/ml.tex

```diff
@@ -2,7 +2,68 @@ \section{AI and Machine Learning Contributions}\label{ml-sec}
+The final CNN was evaluated on the test set, reaching prediction accuracy of 99.7\%, which means that the model mis-classified around 66k of the 22 million implications. Since this accuracy was somewhat surprising for such a small and “simple” model, as a control we generated a random label (yes/no) for each pair of equations, then trained the CNN on this data with 60\%/20\%/20\% training/validation/test percentages, resulting in a 49.99\% accuracy, as expected from a nonbiased model.
+
+\smallskip
+
+It could be the case that the high accuracy of our CNN model was mostly due to it learning the transitivity of the implication relation, as opposed to it discovering patterns in the identities. To clarify this point, it was proposed to train our CNN model either on a random poset or on the equational poset with vertices and labels permuted, and check whether a similar accuracy was achieved. This experiment has not been performed yet.
+
+\smallskip
+
+In any case, since there are around 600k explicit implications from which the rest can be derived by transitivity (2.7\% of the total), if the CNN was learning transitivity it should perform well with a really small training dataset. Accordingly, we trained and assessed a CNN model with a 5\%/5\%/90\% training/validation/test proportion, with a resulting 99.6\% accuracy on test (the process finishing in under 20 minutes). But the high accuracy is maintained with even smaller training datasets, as evidenced in the following table:
+\begin{table}[h]
+\centering
+\begin{tabular}{|c|c|}
+\hline
+  % after \\: \hline or \cline{col1-col2} \cline{col3-col4} ...
+  \textbf{Training/validation/test proportion (\%)} & \textbf{Prediction accuracy (\%)} \\\hline\hline
+  60/20/20 & 99.7 \\\hline
+  5/5/90 & 99.6 \\\hline
+  1/1/98 & 99.3 \\\hline
+  0.5/0.5/99 & 98.9 \\\hline
+  0.1/0.1/99.8 & 92.2 \\
+  \hline
+\end{tabular}
+\caption{Prediction accuracy as function of size of training set}
+\end{table}
+
+\noindent Since these training datasets were sensibly smaller than the subset of explicit implications, and were not carefully chosen from the poset extremes but taken randomly, we can conclude that even if the CNN was learning transitivity, probably this alone is not enough to explain the high accuracy achieved by the CNN model.
```

Comment:
maybe reword "sensibly"?  If you're emphasizing the size, just use "much", or if you want to emphasize that the training datasets by nature had to be smaller than the implication dataset, maybe "naturally".  

Also, maybe "were learning", and "that alone is unlikely to" or "that by itself is probably insufficient" or something, adverbs of probability sometimes sound marked at the start of a clause.